### ISSUE TITLE
react-select@1.2.3 does not appear to work with react-dom@16.2.0

### ISSUE THREAD

**Issue by panesofglass** [JedWatson/react-select#2424]
I just updated my project to use React 16.2.0. Almost everything seems to work except some of my `Select` components. If I have a `Select` component with only one placeholder item, attempting to select the component throws an error and breaks the application completely. If I have more than the first element, it's fine. While debugging into `react-select`, I discovered the error comes a `null` `focusedOptionNode`.

<img width="812" alt="screen shot 2018-03-16 at 4 34 05 pm" src="https://user-images.githubusercontent.com/8803/37548829-f0b3b5c4-2937-11e8-81ad-ad5ed22be59d.png">

I tried upgrading to react-select@2.0.0-alpha6, and while the issue is addressed there, I don't see the label string displayed after I select an option.

The following plnkr is a decent representation of the first case of using React 16 with react-select@1.2.1, except that I can't get the plnkr to render anything at all.

https://plnkr.co/edit/OzfJLcqIRmpI2QvjJSJF?p=preview

**Comment by AkosLukacs** [comment]
I guess prop-types script reference was wrong that's why the plunk did not render, Here is a working version: https://plnkr.co/edit/hchRNZ2CC2NYwN9HoKmf?p=preview
Selection seems to be working on this example.

**Comment by panesofglass** [comment]
Thanks, @AkosLukacs. That plunk works, though it uses `react-select@1.2.1`. When I switch to `react-select@1.2.3`, as identified above, it fails to render.

**Comment by AkosLukacs** [comment]
But there is no v.1.2.3. At least the releases page here and npm says v.1.2.1 is the lates.
If I open https://unpkg.com/react-select@1.2.3/dist/react-select.js in the browser, it says: `Cannot find package react-select@1.2.3`

**Comment by panesofglass** [comment]
@AkosLukacs, then that probably explains my issue. :) I would have sworn we had v1.2.3 in our yarn.lock file.

**Comment by panesofglass** [comment]
@AkosLukacs thanks! Now I see I was looking at the version of the typescript declarations. We are on 1.2.1. I'll see if I can reproduce the issue in the punk.

**Comment by panesofglass** [comment]
In the middle of trying to fix the plunk, I discovered that someone elsewhere in our code was filtering out the empty value from the options list, causing the value of `-1` to no longer match an option. So it seems this is not an issue with `react-select`. I'm surprised, though, that we did not see this issue before upgrading to `1.2.1`. Nevertheless, I'll close this issue. Thanks for your help, @AkosLukacs.
